### ISSUE TITLE
Indent level for fat arrow functions

### ISSUE THREAD

**Issue by CSchulz** [eslint/eslint#8575]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 6.9.5
* **npm Version:** 3.10.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
{
    "extends": "standard",
    "plugins": [
        "standard",
        "promise"
    ]
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
describe('AuthenticationService', () => {
  describe('login', () => {
    it('should not be able to login with wrong credentials',
       inject([AuthenticationService, MockBackend], () => {
         failCheck(
           () => service.login('admin', 'admin'),
         );
       },
    ));

    it('should be able to login with correct credentials',
       inject([AuthenticationService, SessionService, MockBackend],
              () => {
         successCheck(
           () => service.login('admin', 'admin'),
         );
      }),
    );
  });
});
```

**What did you expect to happen?**
The `successCheck` should be correctly indented.

It would be nice if the line break resets the indent to the value of `inject` + 2 instead of orientating at the indent level of the parameter.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  14:10  error  Expected indentation of 16 spaces but found 9  indent
  17:7   error  Expected indentation of 14 spaces but found 6  indent
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, this is working as intended -- it's much more common to offset the body of the arrow function from the arrow and the function's parameters, rather than picking an indentation based on something outside of the arrow function (e.g. whether it's a parameter in an `inject` function call).

**Comment by CSchulz** [comment]
Okay. Would it be possible to add a flag for something like that?

If possible, I would also help with implementing it.

**Comment by platinumazure** [comment]
@CSchulz Does this generalize to a slightly different concept: "If I indent a line 2 indent levels instead of 1 from the previous line, I expect that line to be regarded as a 'continuation' and the next line should only be indented 1 level unless other intervening tokens demand a different indentation level"?

**Comment by not-an-aardvark** [comment]
@CSchulz In theory, yes. However, we generally only add options to rules if they meet a set of [guidelines](http://eslint.org/docs/developer-guide/contributing/new-rules#core-rule-guidelines), such as being generally applicable to many developers. (We can't add options for every possible preference, because maintaining all the options would impose a large maintenance burden .) It's possible that this option meets those criteria, but I'm not sure whether it does.

Could you elaborate on what the flag would do? How would it affect arrow function indentations in general? For example, would it affect arrow functions that are not function arguments?

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended, but feel free to create a new issue if you'd like to make a proposal for a new option.
